packages/fetchai/connections/soef/connection.py

```diff
@@ -393,10 +423,22 @@ async def send(self, envelope: Envelope) -> None:
     async def _request_text(self, *args: Any, **kwargs: Any) -> str:
         """Perform and http request and return text of response."""
 
-        def _do_request() -> str:
-            return requests.request(*args, **kwargs).text
+        def _do_request() -> requests.Response:
+            return requests.request(*args, **kwargs)
 
-        return await self.loop.run_in_executor(self._executor_pool, _do_request)
+        try:
+            response = await self.loop.run_in_executor(self._executor_pool, _do_request)
+        except requests.ConnectionError as e:
+            raise SOEFNetworkConnectionError(e) from e
+
+        if response.status_code < 200 or response.status_code >= 300:
+            raise SOEFServerBadResponseError(
+                f"Bad server response: code {response.status_code} when 2XX expected."
```

Comment:
Should we also print response text? and the request (so we can associate)

Reply:
done